File: src/types/specialAbility/sub/Brew.ts
/**
 * @main Brew
 */

/**
 * @title Brew
 */
export type Brew = {
  /**
   * The brew's identifier. An unique, increasing integer.
   * @integer
   * @minimum 1
   */
  id: number

  /**
   * All translations for the entry, identified by IETF language tag (BCP47).
   * @minProperties 1
   */
  translations: {
    /**
     * @patternProperties ^[a-z]{2}-[A-Z]{2}$
     */
    [localeId: string]: {
      /**
       * The brew name.
       * @minLength 1
       */
      name: string
    }
  }
}

File: Sources/OptolithDatabaseSchema/GeneratedTypes/traditionArtifacts/sub/Brew.swift
//
//  Brew.swift
//  OptolithDatabaseSchema
//

import DiscriminatedEnum

public struct Brew: LocalizableEntity {
    /// The brew's identifier. An unique, increasing integer.
    public let id: Int

    /// All translations for the entry, identified by IETF language tag (BCP47).
    public let translations: LocaleMap<BrewTranslation>

    public init(id: Int, translations: LocaleMap<BrewTranslation>) {
        self.id = id
        self.translations = translations
    }
}

public struct BrewTranslation: EntitySubtype {
    /// The brew name.
    public let name: NonEmptyString

    public init(name: NonEmptyString) {
        self.name = name
    }
}
